back/app/services/participation_permissions_service.rb

```diff
@@ -1,8 +1,10 @@
 # frozen_string_literal: true
 
-class ParticipationContextService
+class ParticipationPermissionsService < PermissionsService
   POSTING_DISABLED_REASONS = {
     project_inactive: 'project_inactive',
+    project_only_visible_to_admins: 'project_only_visible_to_admins',
+    project_only_visible_to_groups: 'project_only_visible_to_groups',
```

Comment:
Do we need to make the distinction between both? Otherwise I think I would have a slight preference for a `project_not_visible` reason, but this is definitely fine.